esx_service/utils/kvESX.py

```diff
@@ -42,130 +47,156 @@
 
-    lib.ObjLib_Pwrite.argtypes = [c_uint32, c_void_p, c_uint64, c_uint64]
-    lib.ObjLib_Pwrite.restype = c_int32
-
-    lib.DiskLib_DBGet.argtypes = [c_uint32, c_char_p, POINTER(c_char_p)]
+    # Define result types for disk lib apis.
+    lib.DiskLib_OpenWithInfo.restype = int
+    lib.DiskLib_Close.restype = int
+    lib.DiskLib_SidecarOpen.restype = int
+    lib.DiskLib_SidecarClose.restype = int
+    lib.DiskLib_SidecarMakeFileName.restype = c_char_p
     lib.DiskLib_DBGet.restype = c_uint32
-
-    lib.DiskLib_DBSet.argtypes = [c_uint32, c_char_p, c_char_p]
     lib.DiskLib_DBSet.restype = c_uint32
 
-    return None
+    return
 
+def get_uint(n):
+   if is_64bits:
+      return c_uint64(n)
+   else:
+      return c_uint32(n)
+
+def disk_is_valid(dhandle):
+   if is_64bits:
+      return dhandle != c_uint64(0)
+   else:
+      return dhandle != c_uint32(0)
 
 # Open a VMDK given its path, the VMDK is opened locked just to
 # ensure we have exclusive access and its not already in use.
 def volOpenPath(volpath):
-    dhandle = c_uint32(0)
-    ihandle = c_uint32(0)
+    dhandle = get_uint(0)
+    ihandle = get_uint(0)
     key = c_uint32(0)
 
-    res = lib.DiskLib_OpenWithInfo(volpath, VMDK_OPEN_FLAGS, byref(key),
+    res = lib.DiskLib_OpenWithInfo(volpath.encode(), VMDK_OPEN_FLAGS, byref(key),
                                    byref(dhandle), byref(ihandle))
 
     if res != 0:
         logging.warning("Open %s failed - %x", volpath, res)
 
     return dhandle
 
-
 # Create the side car for the volume identified by volpath.
 def create(volpath, kv_dict):
-    disk = c_uint32(0)
-    objHandle = c_uint32(0)
-
-    disk = volOpenPath(volpath)
-
-    if disk == c_uint32(0):
-        return False
-
-    if useSideCarCreate:
-        res = lib.DiskLib_SidecarCreate(disk, DVOL_KEY, KV_CREATE_SIZE,
-                                        KV_SIDECAR_CREATE, byref(objHandle))
-    else:
-        res = lib.DiskLib_SidecarOpen(disk, DVOL_KEY, KV_SIDECAR_CREATE,
-                                      byref(objHandle))
+    objHandle = get_uint(0)
+    dhandle = volOpenPath(volpath)
 
-    if res != 0:
-        logging.warning("Side car create for %s failed - %x", volpath, res)
-        lib.DiskLib_Close(disk)
-        return False
+    if disk_is_valid(dhandle):
+       if useSideCarCreate:
+          res = lib.DiskLib_SidecarCreate(dhandle, DVOL_KEY.encode(), KV_CREATE_SIZE,
+                                          KV_SIDECAR_CREATE, byref(objHandle))
+       else:
+          res = lib.DiskLib_SidecarOpen(dhandle, DVOL_KEY.encode(), KV_SIDECAR_CREATE,
+                                        byref(objHandle))
+       if res != 0:
+          logging.warning("Side car create for %s failed - %x", volpath, res)
+          lib.DiskLib_Close(dhandle)
+          return False
 
-    lib.DiskLib_SidecarClose(disk, DVOL_KEY, byref(objHandle))
-    lib.DiskLib_Close(disk)
+       lib.DiskLib_SidecarClose(dhandle, DVOL_KEY.encode(), byref(objHandle))
```

Comment:
Are these indentations correct? I'm not sure they aren't, but it looks like they changed to move under
 `if disk_is_valid(dhandle):`

In this case there is no other return value when the disk is invalid and None will be returned. I'm not sure if this is correct or not since there is no specification of the return value or any doc comments explaining what is happening.